src/Compilers/Test/Utilities/CSharp/FunctionPointerUtilities.cs

```diff
@@ -128,6 +130,42 @@ static void verifyParameter(ParameterSymbol symbol, MethodSymbol containing)
             }
         }
 
+        public static void VerifyPublicFunctionPointerGetTypeInfo(SemanticModel model, SyntaxNode syntax, bool useConvertedType,
+            Action<ITypeSymbol> returnTypeVerifier,
+            params Action<ITypeSymbol>[] paramVerifiers)
+        {
+            var typeInfo = model.GetTypeInfo(syntax);
+            var typeSymbol = (IFunctionPointerTypeSymbol)(useConvertedType ? typeInfo.ConvertedType : typeInfo.Type)!;
+
+            if (!useConvertedType)
+            {
+                Assert.Equal(typeInfo.Type, typeInfo.ConvertedType, SymbolEqualityComparer.IncludeNullability);
+            }
+
+            CommonVerifyFunctionPointer((FunctionPointerTypeSymbol)((CSharp.Symbols.PublicModel.FunctionPointerTypeSymbol)typeSymbol).UnderlyingTypeSymbol);
+
+            returnTypeVerifier(typeSymbol.Signature.ReturnType);
+
+            Assert.Equal(paramVerifiers.Length, typeSymbol.Signature.Parameters.Length);
+
+            foreach (var (verifier, parameter) in paramVerifiers.Zip(typeSymbol.Signature.Parameters, (v, p) => (v, p)))
```

Comment:
>foreach (var (verifier, parameter) in paramVerifiers.Zip(typeSymbol.Signature.Parameters, (v, p) => (v, p))) [](start = 12, length = 108)

Why so complicated? Can't we just do a for based on index?

Reply:
I don't particularly find `for` loops where I have to manually keep track of indexes simpler than a `Zip` monda.

---
In reply to: [428900604](https://github.com/dotnet/roslyn/pull/44436#discussion_r428900604) [](ancestors = 428900604)